### DISCUSSION TITLE
DOSXYZnrc error "The job has failed"

### DISCUSSION THREAD

**Discussion by phantasuson** [nrc-cnrc/EGSnrc#791] [Questions] [answered]
DOSXYZnrc always reports error, neither the phantom I built nor example from other can run. I don't know if it's my misoperation or a bug.
![image](https://user-images.githubusercontent.com/80371548/138408126-e51588be-c458-48d4-a986-c7941584dfb9.png)
![image](https://user-images.githubusercontent.com/80371548/138410608-426ae6b0-9273-41b8-bed7-9d7d72ac4f51.png)

**Comment by rtownson** [comment, marked as answer]
Could you try running this on the command line, to see if it gives a more informative error? E.g.

```
dosxyznrc -i 6MV_Water -p 700icru
```

Make sure that the input file is in `egs_home/dosxyznrc`. What is the output you see when you try this?

**Reply by phantasuson** [reply]
Thanks! It works after dragging the input file to dosxyznrc. I was going to put it into a folder for classifying. But there are only `.egslst` and `.errors` in output files list. What should I do if I want .3ddose file? By the way, I saw others get the PDD data in `.egslst` file, but there is no such data in my `.egslst` file although have selected the voxels to list the dose. Am I missing something? The log is below:
[新建文本文档 (2).txt](https://github.com/nrc-cnrc/EGSnrc/files/7418398/2.txt)

**Reply by rtownson** [reply]
It looks like it's failing to open and read the phase-space `D:/EGSnrc/EGSnrc/egs_home/BEAM_TEST2/TEST2.egsphsp3`, are you sure this file exists and has particles in it?

**Reply by phantasuson** [reply]
I‘ve fixed the error above, it is ZLAST does not match. I did a successful run, but I didn't get ideal PDD data. What could be the cause?
[log.txt](https://github.com/nrc-cnrc/EGSnrc/files/7425500/log.txt)
[10-27-3egslst.txt](https://github.com/nrc-cnrc/EGSnrc/files/7425506/10-27-3egslst.txt)

**Reply by rtownson** [reply]
The PDD is in the .egslst file, under `DOSXYZnrc Dose outputs`. Right now it's mostly zero just because you haven't run very many histories, it will require a much longer simulation. Note that you can also extract a PDD from a 3ddose file.

**Reply by phantasuson** [reply]
I ran 10^7 histories but I still didn't get the data we wanted. There are 100 voxels along the z axis I've selected to list the dose but only the first 40 were listed. It said `The dose values in 5538 voxels had error > 50% and have been zeroed` .Is histories still not enough?
[10-28.egslst.txt](https://github.com/nrc-cnrc/EGSnrc/files/7430832/10-28.egslst.txt)

**Reply by phantasuson** [reply]
10^8 histories running perform almost the same as 10^7...... x_x

**Reply by rtownson** [reply]
The reason the uncertainty is not reducing is because you are just recycling the same particles, so they are not statistically independent histories. I see in the egslst file you sent that you are already recycling 52x, that is too many (usually 2-10 is reasonable). Try generating a larger phase-space so that you can reduce the recycling (and uncertainty).

**Reply by phantasuson** [reply]
I've got this part figured out. Thank you for your patience all the time.

**Reply by rtownson** [reply]
What was the solution?

**Reply by phantasuson** [reply]
I regenerated a 10^9 histories phase-space. Meanwhile I realized that I had been mistaking uncertainty for PDD
